docs/getting-started/components/registry.md

```diff
@@ -1,31 +1,47 @@
 
-Each Feast deployment has a single feature registry. Feast only supports file-based registries today, but supports four different backends.
+Feast comes with built-in file-based and sql-based registry implementations. By default, Feast uses a file-based registry, which stores the protobuf representation of the registry as a serialized file in the local file system. For more details on which registries are supported, please see [Registries](../../reference/registries/).
 
-* `Local`: Used as a local backend for storing the registry during development
-* `S3`: Used as a centralized backend for storing the registry on AWS
-* `GCS`: Used as a centralized backend for storing the registry on GCP
-* `[Alpha] Azure`: Used as centralized backend for storing the registry on Azure Blob storage.
+## Updating the registry
 
-The feature registry is updated during different operations when using Feast. More specifically, objects within the registry \(entities, feature views, feature services\) are updated when running `apply` from the Feast CLI, but metadata about objects can also be updated during operations like materialization.
+We recommend users store their Feast feature definitions in a version controlled repository, which then via CI/CD
+automatically stays synced with the registry. Users will often also want multiple registries to correspond to
+different environments (e.g. dev vs staging vs prod), with staging and production registries with locked down write
+access since they can impact real user traffic. See [Running Feast in Production](../../how-to-guides/running-feast-in-production.md#1.-automatically-deploying-changes-to-your-feature-definitions) for details on how to set this up.
 
-Users interact with a feature registry through the Feast SDK. Listing all feature views:
+## Accessing the registry from clients
+
+Users can specify the registry through a `feature_store.yaml` config file, or programmatically. We often see teams
+preferring the programmatic approach because it makes notebook driven development very easy:
+
+### Option 1: programmatically specifying the registry
 
 ```python
-fs = FeatureStore("my_feature_repo/")
-print(fs.list_feature_views())
+repo_config = RepoConfig(
+    registry=RegistryConfig(path="gs://feast-test-gcs-bucket/registry.pb"),
+    project="feast_demo_gcp",
+    provider="gcp",
+    offline_store="file",  # Could also be the OfflineStoreConfig e.g. FileOfflineStoreConfig
+    online_store="null",  # Could also be the OnlineStoreConfig e.g. RedisOnlineStoreConfig
+)
+store = FeatureStore(config=repo_config)
 ```
 
-Or retrieving a specific feature view:
+### Option 2: specifying the registry in the project's `feature_store.yaml` file
 
-```python
-fs = FeatureStore("my_feature_repo/")
-fv = fs.get_feature_view(“my_fv1”)
+```yaml
+project: feast_demo_aws
+provider: aws
+registry: s3://feast-test-s3-bucket/registry.pb
+online_store: null
+offline_store:
+  type: file
 ```
 
-{% hint style="info" %}
```

Comment:
Why take out this hint? I think it's useful.

Reply:
Because this only mostly applies to file-based registries, sql registry used protobuf but isn't a itself a protobuf object. I suppose I can add this back with some caveats mentioned.

Reply:
Thank you!